Code/Framework/AzToolsFramework/Platform/Linux/AzToolsFramework/API/PythonLoader_Linux.cpp

```diff
@@ -14,7 +14,7 @@ namespace AzToolsFramework::EmbeddedPython
 {
     PythonLoader::PythonLoader()
     {
-        constexpr char libPythonName[] = "libpython3.7m.so.1.0";
+        constexpr char libPythonName[] = "libpython3.10.so.1.0";
```

Comment:
Can extract this file path from the `3rdParty::Python` imported target and add it as a COMPILER_DEFINITION to the AzToolsFramework target, so that we don't need to specify the python library in code?

Reply:
That would be great, but initial testing looks like if I query for the INTERFACE_LINK_LIBRARIES property, it returns a generator expression for the value

```
$<IF:$<CONFIG:Debug>,/home/github/o3de/python/runtime/python-3.10.5-rev1-linux/python/lib/libpython3.10.so.1.0,/home/github/o3de/python/runtime/python-3.10.5-rev1-linux/python/lib/libpython3.10.so.1.0>
```

And linux is the only platform that we are doing this preload of the python shared library on

Reply:
The [target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html?highlight=target_compile_definitions) function that gets invoked internally in ly_add_target supports generator expressions.

Reply:
I was getting this error:
```
target_compile_definitions called with invalid arguments
```
with this platform include cmake
```
get_target_property(libraries 3rdParty::Python INTERFACE_LINK_LIBRARIES)

set(LY_COMPILE_DEFINITIONS PYTHON_SHARED_LIBRARY_PATH="${libraries}")
```

Reply:
Done